### DISCUSSION TITLE
How to add cache to current_active_user

### DISCUSSION THREAD

**Discussion by dima23113** [fastapi-users/fastapi-users#1387] [Q&A] [answered]
Hello! How to add cache to current active user? I'm a little worried that every visit to a protected endpoint makes a request to the database

**Comment by frankie567** [comment, marked as answer]
Unless you heavily modify the [Authenticator class](https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/authenticator.py), that's not easily doable.

That said, you shouldn't be too worried about the DB query. It's fast and guarantees you maximum security (since we check at each request the existence and state of the user). 99% of web frameworks out there are doing it like this.

**Reply by dima23113** [reply]
Thank you for the answer!
